bundles/org.openhab.binding.siemenshvac/src/main/java/org/openhab/binding/siemenshvac/internal/metadata/SiemensHvacMetadataRegistryImpl.java

```diff
@@ -0,0 +1,1270 @@
+            dptMap.put("byDptId" + dpi.getDptId(), node);
+        }
+    }
+
+    class ResolveCount {
+        private int resolveCount = 0;
+
+        public ResolveCount(int count) {
+            resolveCount = count;
+        }
+
+        public void decreaseResolveCount() {
+            resolveCount--;
+        }
+
+        public int getResolveCount() {
+            return resolveCount;
+        }
+    }
+
+    public void resolveDetails(int unresolveCountP) {
+        ResolveCount rv = new ResolveCount(unresolveCountP);
+
+        for (String key : dptMap.keySet()) {
+            if (key.indexOf("byId") < 0) {
+                continue;
+            }
+
+            SiemensHvacMetadata node = dptMap.get(key);
+            if (node != null) {
+                if (node.getClass() == SiemensHvacMetadataDataPoint.class) {
+                    SiemensHvacMetadataDataPoint dpi = (SiemensHvacMetadataDataPoint) node;
+                    if (!dpi.getDetailsResolved()) {
+                        resolveDptDetails(dpi, rv);
+                    }
+                }
+            }
+        }
+    }
+
+    public int unresolveCount() {
+        int count = 0;
+        for (String key : dptMap.keySet()) {
+            if (key.indexOf("byId") < 0) {
+                continue;
+            }
+
+            SiemensHvacMetadata node = dptMap.get(key);
+            if (node != null) {
+                if (node.getClass() == SiemensHvacMetadataDataPoint.class) {
+                    SiemensHvacMetadataDataPoint dpi = (SiemensHvacMetadataDataPoint) node;
+                    if (!dpi.getDetailsResolved()) {
+                        count++;
+                    }
+                }
+            }
+
+        }
+
+        return count;
+    }
+
+    @Override
+    public @Nullable SiemensHvacMetadataMenu getRoot() {
+        return (SiemensHvacMetadataMenu) root;
+    }
+
+    @Override
+    public void readMeta() throws SiemensHvacException {
+        ArrayList<SiemensHvacMetadataDevice> lcDevices = devices;
+        SiemensHvacConnector lcHvacConnector = hvacConnector;
+
+        if (root != null) {
+            return;
+        }
+
+        if (lcHvacConnector == null) {
+            logger.debug("SiemensHvacMetadataRegistryImpl:ReadMeta() : lHvacConnector not initialize.");
+            return;
+        }
+
+        readUserInfo();
+
+        SiemensHvacBridgeConfig config = lcHvacConnector.getBridgeConfiguration();
+        if (config == null) {
+            logger.debug("SiemensHvacMetadataRegistryImpl:ReadMeta() : config not initialize.");
+            return;
+        }
+
+        SiemensHvacMetadataUser user = null;
+
+        String userName = config.userName;
+        if (userList.containsKey(userName)) {
+            user = userList.get(userName);
+        }
+
+        if (user == null) {
+            logger.error("SiemensHvacMetadataRegistryImpl:ReadMeta() : cannot find user, aborting.");
```

Comment:
Yes, this is fatal error, as at this point, binding will not be able to read metadata, and continue normal operations.
I change this line to throw an exception instead of logging an error, like this the binding will go offline.
Fix on next commit